File: patterns/long-press/rules-js/long-press-fallback.js
rules.JSRule({
  name: "Button gedrueckt (mit Fallback)",
  triggers: [triggers.ItemStateChangeTrigger("ButtonState", "OFF", "ON")],
  execute: (event) => {
    actions.ScriptExecution.createTimer(time.ZonedDateTime.now().plusSeconds(5), () => {
      if (items.getItem("ButtonState").state === "ON") {
        console.log("Fallback: kein OFF-Signal erhalten, setze ButtonState manuell zurueck");
        items.getItem("ButtonState").postUpdate("OFF");
      }
    });
  }
});

File: patterns/long-press/rules-python/long_press_fallback.py
import threading

from datetime import datetime

from openhab import rule, Registry, logger
from openhab.triggers import ItemStateChangeTrigger


def fallback_off():
    if str(Registry.getItem("ButtonState").getState()) == "ON":
        logger.info("Fallback: kein OFF-Signal erhalten, setze ButtonState manuell zurueck")
        Registry.getItem("ButtonState").postUpdate("OFF")


@rule(triggers=[ItemStateChangeTrigger("ButtonState", state="ON", previous_state="OFF")])
class ButtonGedruecktMitFallback:
    def execute(self, module, input):
        threading.Timer(5, fallback_off).start()
